lib/slackiq.rb

```diff
@@ -1,156 +1,192 @@
-require "slackiq/version"
-
 require "net/http"
+require "uri"
 require "json"
-require "httparty"
-
-require "slackiq/time_helper"
-
-require "active_support" # For Hash#except
-
-module Slackiq
+require "date"
 
-  class << self
+class Slackiq
 
-    # Configure all of the webhook URLs you're going to use
-    # @author Jason Lew
-    def configure(webhook_urls={})
-      raise "Argument must be a Hash" unless webhook_urls.class == Hash
-      @@webhook_urls = webhook_urls
-    end
-
-    # Send a notification to Slack with Sidekiq info about the batch
-    # @author Jason Lew
-    def notify(options={})
-      url = @@webhook_urls[options[:webhook_name]]
-      title = options[:title]
-      # description = options[:description]
-      status = options[:status]
-
-      if (bid = options[:bid]) && status.nil?
-        raise <<~EOT.chomp unless defined?(Sidekiq::Batch::Status)
-                 Sidekiq::Batch::Status is not defined. \
-                 Are you sure Sidekiq Pro is set up correctly?
-                 EOT
-        status = Sidekiq::Batch::Status.new(bid)
-      end
+  VERSION = "1.2.1".freeze
 
-      color = options[:color] || color_for(status)
+  attr_reader :options
 
-      extra_fields = options.except(:webhook_name, :title, :description, :status)
+  # @param options [Hash]
+  def self.notify(options)
+    new(options).execute
+  end
 
-      fields = []
+  # @param options [Hash]
+  def self.configure(webhook_urls={})
+    @@webhook_urls = webhook_urls
```

Comment:
Rarely does one actually need "class variables". One `@` in a `self` context would signify it's a class instance variable and that is generally sufficient.